server/service/src/programs/contact_trace/contact_trace_updated.rs

```diff
@@ -0,0 +1,90 @@
+use super::{
+    contact_trace_schema::{SchemaContactTrace, SchemaGender},
+    upsert::UpsertContactTraceError,
+};
+use chrono::{DateTime, NaiveDate};
+use repository::{
+    contact_trace::{ContactTraceFilter, ContactTraceRepository},
+    contact_trace_row::{ContactTraceRow, ContactTraceRowRepository},
+    Document, Gender, ProgramRow, StorageConnection, StringFilter,
+};
+use std::str::FromStr;
+use util::hash::sha256;
+
+/// Callback called when a program enrolment document has been updated
+pub(crate) fn update_contact_trace_row(
+    con: &StorageConnection,
+    patient_id: &str,
+    document: &Document,
+    contact_trace: SchemaContactTrace,
+    program_row: ProgramRow,
+) -> Result<ContactTraceRow, UpsertContactTraceError> {
+    let datetime = DateTime::parse_from_rfc3339(&contact_trace.datetime)
+        .map_err(|err| {
+            UpsertContactTraceError::InternalError(format!(
+                "Invalid enrolment datetime format: {}",
+                err
+            ))
+        })?
+        .naive_utc();
+
+    // Retrieve existing or create now contact trace id
+    let repo = ContactTraceRepository::new(con);
+    let contact_trace_row = repo
+        .query_by_filter(ContactTraceFilter {
+            document_name: Some(StringFilter::equal_to(&document.name)),
+            ..ContactTraceFilter::default()
+        })?
+        .pop();
+    let contact_trace_id = match contact_trace_row {
+        Some(contact_trace_row) => contact_trace_row.0.id,
```

Comment:
Because these comments where about graphql as far as I remember ;-) This was mostly copy and past from other repos and its the underlying diesel join... However, agree tuples are not that nice here will update.

Reply:
> Because these comments where about graphql as far as I remember ;-) 

Yeah I think they were only made in graphql context, but they were made as general comment.

> This was mostly copy and past from other repos

Ahh ok, i must have missed those other repos in PR review, since when we started using 'domain' object as composition of repository rows, we used struct, I thought this pattern continued. Thanks for change